src/Optical/OpticalSystem.jl

```diff
@@ -45,14 +45,13 @@ CSGOpticalSystem(
 """
 struct CSGOpticalSystem{T,D<:Number,S<:Surface{T},L<:LensAssembly{T}} <: AbstractOpticalSystem{T}
     assembly::L
-    detector::S
-    detectorimage::HierarchicalImage{D}
+    detector::Pair{S, AbstractDetector{D}}
     temperature::T
     pressure::T
 
     function CSGOpticalSystem(
         assembly::L,
-        detector::S,
+        detector::Pair{S, String},
```

Comment:
Don't understand what this constructor would be used for.

Reply:
To give detectors a name, once multiple detectors are implemented, so you can refer to detectors by name later, or use the name for reporting